jsonrpc/filter_manager.go

```diff
@@ -227,23 +228,22 @@ func (f *FilterManager) Run() {
 }
 
 func (f *FilterManager) nextTimeoutFilter() *Filter {
-	f.lock.Lock()
+	f.filtersLock.RLock()
```

Comment:
I think it just complicates the logic having to lock the entire filters manager to read the size of the filters. What do you think about having it as an atomic counter?

Reply:
What do you mean? There will be race conditions for timers as well. Should lock whole method

Reply:
Aha, so we need to hold the lock while fetching an item from the `f.timer`?

Reply:
Yes

Reply:
Closing because I refactored the code. Please review again 🙏